Homework11/WebTableTests.cs

```diff
@@ -0,0 +1,157 @@
+            var salaryInput = _driver.FindElement(By.XPath("//input[@id='salary']"));
+            var departmentInput = _driver.FindElement(By.XPath("//input[@id='department']"));
+            var submitButton = _driver.FindElement(By.XPath("//button[@id='submit']"));
+
+            // Fill in form inputs.
+            FillInInput(fistNameInput, Constants.FirstName);
+            FillInInput(lastNameInput, Constants.LastName);
+            FillInInput(emailInput, Constants.Email);
+            FillInInput(ageInput, Constants.Age);
+            FillInInput(salaryInput, Constants.Salary);
+            FillInInput(departmentInput, Constants.Department);
+            submitButton.Click();
+
+            // Get text of new entry.
+            ReadOnlyCollection<IWebElement> gridcell = _driver.FindElements(By.XPath("((//div[@role='row'])[5])/div"));
+            Assert.AreEqual(ExpectedConstants.NewEntryValue, GetTextOfCollectionElements(gridcell));
+        }
+
+        // Test4: Edit first entry.
+        [Test]
+        public void EditFirstEntry()
+        {
+            var editButton = _driver.FindElement(By.XPath("//span[@title='Edit']"));
+            editButton.Click();
+
+            // Find form inputs.
+            var fistNameInput = _driver.FindElement(By.XPath("//input[@id='firstName']"));
+            var lastNameInput = _driver.FindElement(By.XPath("//input[@id='lastName']"));
+            var emailInput = _driver.FindElement(By.XPath("//input[@id='userEmail']"));
+            var ageInput = _driver.FindElement(By.XPath("//input[@id='age']"));
+            var salaryInput = _driver.FindElement(By.XPath("//input[@id='salary']"));
+            var departmentInput = _driver.FindElement(By.XPath("//input[@id='department']"));
+            var submitButton = _driver.FindElement(By.XPath("//button[@id='submit']"));
+
+            // Clear form inputs.
+            DeleteValue(fistNameInput);
+            DeleteValue(lastNameInput);
+            DeleteValue(emailInput);
+            DeleteValue(ageInput);
+            DeleteValue(salaryInput);
+            DeleteValue(departmentInput);
+
+            // Fill in form inputs.
+            FillInInput(fistNameInput, Constants.FirstName);
+            FillInInput(lastNameInput, Constants.LastName);
+            FillInInput(emailInput, Constants.Email);
+            FillInInput(ageInput, Constants.Age);
+            FillInInput(salaryInput, Constants.Salary);
+            FillInInput(departmentInput, Constants.Department);
+            submitButton.Click();
+
+            // Get text of updated entry.
+            ReadOnlyCollection<IWebElement> gridcell = _driver.FindElements(By.XPath("((//div[@role='row'])[2])/div"));
+            Assert.AreEqual(ExpectedConstants.NewEntryValue, GetTextOfCollectionElements(gridcell));
+        }
+
+        // Test5: Delete first entry.
+        [Test]
+        public void DeleteFirstEntry()
```

Comment:
Ok, I can apply this tests but here can be problem in common approach:
Again - you know that you will see here row with name "Vega" and first found button with title='Delete' it's a button to delete exactly this row. But what if you create data for tests...
The correct steps - found the first row and get the name in the first row -> get delete button (locator) exactly for the first row (or row with name you've found a step above) -> delete row -> get all names from rows and check there are no deleted name

Reply:
fixed